### ISSUE TITLE
`preserveLatex` doesn't preserve LaTeX in `3x+y`

### ISSUE THREAD

**Issue by bengolds** [cortex-js/compute-engine#29]
The `preserveLaTeX` option works _most_ of the time, but it misses preserving the LaTeX at a level sometimes. For example, in the expression `3x+y`, you get:

```javascript
{
  latex: '3x+y',
  fn: [
    'Add',
    [ 'Multiply', { num: '3', latex: '3' }, { latex: 'x', sym: 'x' } ], // This is short form! We'd expect long form with latex.
    { latex: 'y', sym: 'y' }
  ]
}

// instead of

{
  latex: '3x+y',
  fn: [
    'Add',
    {
      latex: '3x'
      fn: [ 'Multiply', { num: '3', latex: '3' }, { latex: 'x', sym: 'x' } ]},
    }
    { latex: 'y', sym: 'y' }
  ]
}
```

# Quick Repro:

```bash
node - << 'EOF'
    const util = require('util')
    const CE = require("@cortex-js/compute-engine")
    const myParse = new CE.LatexSyntax({preserveLatex: true})
    console.log(util.inspect(myParse.parse('3x+y'), {depth: null}))
EOF
```

This prints out:

**Comment by arnog** [comment]
@bengolds  Do you expect `latex` to be present even for trivial cases, i.e. when a symbol or number and the latex is the same as the value of the expression?

That is:

`x` → `{ latex: "x", sym: "x"}` or `{ sym: "x" }`

`2` → `{ latex: "2", num: "2"}` or `{ num: "2" }`

**Comment by bengolds** [comment]
For `Symbol`, `StringLiteral`, or `Number`, we don't expect it or store it. We're perfectly content ignoring it :)

For `Function`, we do need it so that we can turn fragments of the `MathJson` back into LaTeX. Without the LaTeX present, we'd basically have to rewrite `ComputeEngine.serialize()` 😅

**Comment by strickinato** [comment]
This appears to still be an issue in `0.4.3`. 

(Repro now with new API)

```
node - << 'EOF'                                                                                                                                         ~/work/compute-engine
    const util = require('util')
    const CE = require("@cortex-js/compute-engine")
    const myParse = new CE.ComputeEngine().latexSyntax // preserveLatex appears to be default now
    console.log(util.inspect(myParse.parse('3x+y'), {depth: null}))
EOF
```

**Comment by bengolds** [comment]
This is fixed in `0.6.0` if you use:
```
const ce = new ComputeEngine();
ce.jsonSerializationOptions.metadata = ['latex'];
ce.jsonSerializationOptions.shorthands = [];
```
